presto-parser/src/main/java/com/facebook/presto/sql/tree/SetPath.java

```diff
@@ -25,27 +25,27 @@
 public class SetPath
         extends Statement
 {
-    private final List<PathElement> path;
+    private final PathSpecification pathSpecification;
 
-    public SetPath(List<PathElement> path)
+    public SetPath(PathSpecification pathSpecification)
```

Comment:
Same here: just use a single constructor and set the missing arguments to Optional.empty()